worker/src/main/java/gov/cms/ab2d/worker/adapter/bluebutton/ContractAdapterStub.java

```diff
@@ -19,65 +18,116 @@
+        }
+
         return sno;
     }
 
-    private List<String> getFromSampleFile(final int startOffset) {
-        if (startOffset < 0) {
+    private List<String> fetchPatientRecords(final int contractSno) {
+        if (contractSno < 0) {
             return new ArrayList<String>();
         }
 
+        int numberOfRows = determineNumberOfRows(contractSno);
+        int rowsToRetrieve = determineRowsToRetrieve(numberOfRows);
+
+        var patientIdRows = getFromSampleFile(rowsToRetrieve);
+        final List<String> allRows = new ArrayList<>();
+        allRows.addAll(patientIdRows);
+
+        if (numberOfRows > MAX_ROWS) {
+            allRows.addAll(createMoreRows(numberOfRows, rowsToRetrieve, patientIdRows));
+        }
+
+        return allRows;
+    }
+
+    private int determineNumberOfRows(int contractSno) {
+        if (contractSno == 0) {
+            return 100;
+        } else {
+            return contractSno * 1000;
+        }
+    }
+
+    private int determineRowsToRetrieve(int numberOfRows) {
+        return numberOfRows < MAX_ROWS ? numberOfRows : MAX_ROWS;
+    }
+
+    private List<String> getFromSampleFile(int rowsToRetrieve) {
+        var patientIdRows = new ArrayList<String>();
+
         try (var inputStream = this.getClass().getResourceAsStream(BENE_ID_FILE)) {
             Assert.notNull(inputStream, "error getting resource as stream :  " + BENE_ID_FILE);
 
             try (var br =  new BufferedReader(new InputStreamReader(inputStream))) {
                 Assert.notNull(br, "Could not create buffered reader from input stream :  " + BENE_ID_FILE);
 
-                return readLinesByOffset(br, startOffset);
+                final List<String> rows = br.lines()
+                        .limit(rowsToRetrieve)
+                        .collect(Collectors.toList());
+
+                patientIdRows.addAll(rows);
             }
         } catch (Exception ex) {
             final String errMsg = "Error reading file : ";
             log.error("{} {} ", errMsg, BENE_ID_FILE, ex);
             throw new RuntimeException(errMsg + BENE_ID_FILE);
         }
+
+        return patientIdRows;
     }
 
-    private List<String> readLinesByOffset(BufferedReader br, int startOffset) {
-        return br.lines()
-                .skip(startOffset)
-                .limit(pageSize)
-                .collect(Collectors.toList());
+    private List<String> createMoreRows(int numberOfRows, int rowsToRetrieve, List<String> patientIdRows) {
+        final int diff = numberOfRows - rowsToRetrieve;
+        if (diff < MAX_ROWS) {
+            return patientIdRows.stream().limit(diff).collect(Collectors.toList());
+        } else {
+            List<String> accumulator = new ArrayList<>();
+            accumulator.addAll(patientIdRows);
+
+            final List<String> moreRows = createMoreRows(diff, rowsToRetrieve, patientIdRows);
+            accumulator.addAll(moreRows);
```

Comment:
Just realized this is a recursion. We are risking `StackOverflowError` in case of large contract numbers. A simple loop would probably do here.

Reply:
done - replaced recursion with a simple loop.